extension/storage/filestorage/README.md

```diff
@@ -93,3 +93,32 @@ processors:
 exporters:
   nop:
 ```
+
+## Feature Gates
+
+See the [Collector feature gates](https://github.com/open-telemetry/opentelemetry-collector/blob/main/featuregate/README.md#collector-feature-gates) for an overview of feature gates in the collector.
+
+### `extension.filestorage.replaceUnsafeCharacters`
+
+When enabled, characters that are not safe in file paths are replaced in component name using the extension before creating the file name to store data by the extension.
+
+This may change the file path that the extension is writing component's data to.
+For example, for component `filelog/logs/json`, the data is stored:
+
+- in path `receiver_filelog_logs/json` with the feature flag disabled (note that this is file named `json` inside directory named `receiver_filelog_logs`).
+- in file `receiver_filelog_logs~json` with the feature flag enabled.
```

Comment:
Wouldn't it be better to encode unsafe characters in terms of safe characters instead of replacing? Replacing creates the possibility of name clashes, e.g. components `filelog/logs/json` and `filelog/logs~json` both convert to `receiver_filelog_logs~json`.

Reply:
That's true, the issue of clashes crossed my mind. Especially with the replacement being quite "eager", replacing many different characters. I'll change the implementation to encode unsafe characters instead of replacing with `~`.

Reply:
@astencel-sumo may I ask if you get any chance to work on this?

Reply:
@newly12 the vacation period is over and I'm going to get back to it.